Update the doc for Activity-Stream dataset (#502)
mozilla/firefox-data-docs: src/datasets/other/activity-stream/reference.md

> # What is Activity Stream?
> 

Reviewer: ```suggestion
# Activity Stream Datasets

This article describes the various BigQuery tables Mozilla uses to store Activity Stream data, along with some examples of how to access them.

## Table of Contents

<!-- toc -->

## What is Activity Stream?

```

* `about:newtab`
* `about:welcome`
    - [starting with Firefox 62](https://bugzilla.mozilla.org/show_bug.cgi?id=1448918)
* `Snippets`
* `CFR`
* `Onboarding`
* `What's new panel`
* `Moments pages`

The Activity Stream team has implemented data collection in and around these pages. This data has some overlap with the standard Firefox Telemetry system, however it is a custom system, designed and maintained by that team.

For specific questions about this data, reach out to the `#fx-messaging-system` Slack channel directly.

## Activity Stream Pings

This data is measured in various custom pings that are sent via [PingCentre](https://searchfox.org/mozilla-central/source/browser/modules/PingCentre.jsm) (different from [Pingsender](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/internals/pingsender.html)).

* [Activity Stream Pings: `data_events.md`](https://firefox-source-docs.mozilla.org/browser/components/newtab/docs/v2-system-addon/data_events.html)
* [Activity Stream Pings: `data_dictionary.md`](https://firefox-source-docs.mozilla.org/browser/components/newtab/docs/v2-system-addon/data_dictionary.html)

## Accessing Activity Stream Data

> Activity Stream pings are stored in the BigQuery database like other metrics of Firefox. It has two namespaces: `activity_stream` and `messaging_system`.

Reviewer: ```suggestion
Activity Stream pings are stored in BigQuery (just like other Firefox Telmetry). There are two datasets containing tables: `activity_stream` and `messaging_system`.
```

> #### Namespace `activity_stream`

Reviewer: ```suggestion
#### `activity_stream`
```

> * Table `events` stores user interactions on the `about:home` and `about:newtab` pages
> * Table `sessions` stores sessions of `about:home` and `about:newtab` pages
> * Table `impression_stats` stores impression/click/block events for the Pocket recommendations on the `about:home` and `about:newtab` pages
> * Table `spoc_fills` stores Pocket Sponsored recommendation related pings

Reviewer: ```suggestion

The `activity_stream` dataset contains the following tables:

* `events` stores user interactions with the `about:home` and `about:newtab` pages
* `sessions` stores sessions of `about:home` and `about:newtab` pages
* `impression_stats` stores impression/click/block events for the Pocket recommendations on the `about:home` and `about:newtab` pages
* `spoc_fills` stores "Pocket Sponsored" recommendation related pings
```

> #### Namespace `messaging_system`

Reviewer: ```suggestion
#### `messaging_system`
```

> 
> * Table `cfr` stores user interactions of the CFR (Contextual Feature Recommendation)
> * Table `moments` stores Moments Pages related pings
> * Table `onboarding` stores user interactions around the Onboarding features
> * Table `snippets` stores impression/click/dismissal of Firefox Snippets
> * Table `whats_new_panel` stores What's New Panel related pings
> * Table `undesired_events` stores system health related events

Reviewer: ```suggestion

The `activity_stream` dataset contains the following tables:

* `cfr` stores metrics on user interactions with the CFR (Contextual Feature Recommendation) system
* `moments` stores "Moments" pages related pings
* `onboarding` stores metrics on user interactions with onboarding features
* `snippets` stores impression/click/dismissal metrics for Firefox Snippets
* `whats_new_panel` stores "What's New" panel related pings
* `undesired_events` stores system health related events
```

## Gotchas and Caveats

> Since this data collection isn't collected or maintained through our standard Telemetry API, there are a number of "gotchas" to keep in mind when working on this data.

Reviewer: ```suggestion
Since this data collection isn't collected or maintained through our standard Telemetry API, there are a number of "gotchas" to keep in mind when working on this data:
```

* **Different ping types in the same table**: The tables in the `activity_stream` namespace can contain multiple types of pings. For example, the `events` table contains both [AS Page Takeover pings](https://firefox-source-docs.mozilla.org/browser/components/newtab/docs/v2-system-addon/data_events.html#page-takeover-ping) as well as [AS User Event pings](https://firefox-source-docs.mozilla.org/browser/components/newtab/docs/v2-system-addon/data_events.html#user-event-pings).

* **Null handling**: Some fields in the Activity Stream data encode nulls with a `'N/A'` string or a `-1` value.

* **Changes in ping behaviors**: These pings continue to undergo development and the behavior as well as possible values for a given ping seem to change over time. For example, older versions of the event pings for clicking on a Topsite do not seem to report `card_types` and `icon_types`, while newer versions do. Caution is advised.

> * **Pocket data**: Data related to Pocket interaction and usage in the `about:home` and `about:newtab` pages get sent to Pocket via this data collection and pipeline. However, due to privacy reasons, the `client_id` is omitted in the ping whenever the Pocket recommendation identifiers are included, instead it reports with another user unique identifier `impression_id`. Though all the Pocket user interactions, such as clicks, dismisses, save to pocket are still reported as the regular events with the `client_id` as long as they don't contain the Pocket recommendation identifiers.

Reviewer: ```suggestion
* **Pocket data**: Data related to Pocket interaction and usage in the `about:home` and `about:newtab` pages get sent to Pocket via this data collection and pipeline. However, due to privacy reasons, the `client_id` is omitted in the ping whenever the Pocket recommendation identifiers are included, instead it reports with another user unique identifier `impression_id`. Though all the Pocket user interactions, such as clicks, dismisses, and "save to pocket" are still reported as the regular events with a `client_id` as long as they don't contain the Pocket recommendation identifiers.
```

## Examples

> #### Sessions per `client_id`

Reviewer: ```suggestion
### Sessions per `client_id`
```

Note: only includes `client_ids` that completed an Activity Stream session that day.

```sql
SELECT
    client_id,
    DATE(submission_timestamp) AS date,
    count(DISTINCT session_id) as num_sessions
FROM
    `moz-fx-data-shared-prod.activity_stream.sessions`
WHERE
    DATE(submission_timestamp) = '20200601'
GROUP BY
    1
```

> #### Topsite clicks and Highlights clicks

Reviewer: ```suggestion
### Topsite clicks and Highlights clicks
```

```sql
SELECT
    client_id,
    DATE(submission_timestamp) AS date,
    session_id,
    page,
    source,
    action_position,
    experiments
FROM
    `moz-fx-data-shared-prod.activity_stream.events`
WHERE
    source in ('TOP_SITES', 'HIGHLIGHTS')
    AND event = 'CLICK'
    DATE(submission_timestamp) = '20200601'
```

> #### Snippet impressions, blocks, clicks, and dismissals

Reviewer: ```suggestion
### Snippet impressions, blocks, clicks, and dismissals
```
